evidence/pool.go

```diff
@@ -193,6 +196,57 @@ func (evpool *Pool) AddEvidence(evidence types.Evidence) error {
 			return fmt.Errorf("failed to verify %v: %w", ev, err)
 		}
 
+		// For potential amnesia evidence, if this node is indicted it shall retrieve a polc
+		// to form AmensiaEvidence
+		if pe, ok := ev.(types.PotentialAmnesiaEvidence); ok {
+			var (
+				height = pe.Height()
+				exists = false
+				polc   types.ProofOfLockChange
+			)
+			pe.HeightStamp = evpool.State().LastBlockHeight
+
+			// a) first try to find a corresponding polc
+			for round := pe.VoteA.Round + 1; round <= pe.VoteB.Round; round++ {
+				polc, exists = evpool.RetrievePOLC(height, round)
+				if exists {
+					// we should not need to verify it if both the polc and potential amnesia evidence have already
+					// been verified. We replace the potential amnesia evidence.
+					err := evpool.AddEvidence(types.MakeAmnesiaEvidence(pe, polc))
+					if err != nil {
+						evpool.logger.Error("Failed to create amnesia evidence from potential amnesia evidence", "err", err)
+						// revert back to processing potential amnesia evidence
+						exists = false
+					}
+					break
+				}
+			}
+
+			// b) check if amnesia evidence can be made now or if we need to enact the trial period
+			if !exists && pe.Primed(1, pe.HeightStamp) {
+				err := evpool.AddEvidence(types.MakeAmnesiaEvidence(pe, types.EmptyPOLC()))
+				if err != nil {
+					return err
+				}
+			} else if !exists && evpool.State().LastBlockHeight+evpool.State().ConsensusParams.Evidence.ProofTrialPeriod <
+				pe.Height()+evpool.State().ConsensusParams.Evidence.MaxAgeNumBlocks {
+				// if we can't find a proof of lock change and we know that the trial period will finish before the
+				// evidence has expired, then we commence the trial period by saving it in the awaiting bucket
+				evBytes := cdc.MustMarshalBinaryBare(pe)
+				key := keyAwaiting(pe)
+				err := evpool.evidenceStore.Set(key, evBytes)
+				if err != nil {
+					return err
+				}
+				// keep track of when the next pe has finished the trial period
```

Comment:
what do you mean by this - like each piece of evidence only has one trial period yes

Reply:
only if evidenceA's trial period is over and it has been upgraded then it will be set to -1 and evidenceB can then advance the `evpool.nextEvidenceTrialEndedHeight`. This variable only tells the evpool when it's time to loop through all the evidence and upgrade the necessary ones